### PR TITLE
Add update and delete methods to invoice item service

### PR DESCRIPTION
**Pull request by subenson**

### CODE REVIEW

**Review by ashkarpetin — COMMENTED** [review]
Please update changelog

**Review by vykhrystiuk — DISMISSED** [review]

File: src/Services/InvoiceItemService.php [hunk 1]
```diff
@@ -99,4 +99,37 @@ public function create($data, $invoiceId, $invoiceItemId = null)
...
+    }
+
+    /**
+     * @param array|JsonSerializable|InvoiceItem $data
+     * @param string $invoiceId
+     * @param string $invoiceItemId
+     *
+     * @return InvoiceItem
+     */
+    public function delete($data, $invoiceId, $invoiceItemId)
```
Comment (VladimirBerdnik) [inline src/Services/InvoiceItemService.php:128]:
```suggestion
    public function delete($invoiceId, $invoiceItemId)
```

File: src/Services/InvoiceItemService.php [hunk 2]
```diff
@@ -99,4 +99,37 @@ public function create($data, $invoiceId, $invoiceItemId = null)
...
+            ['invoiceId' => $invoiceId, 'invoiceItemId' => $invoiceItemId]
+        );
+    }
+
+    /**
+     * @param array|JsonSerializable|InvoiceItem $data
+     * @param string $invoiceId
+     * @param string $invoiceItemId
+     *
+     * @return InvoiceItem
```
Comment (VladimirBerdnik) [inline src/Services/InvoiceItemService.php:126]: We cannot return InvoiceItem as we just deleted it

File: src/Services/InvoiceItemService.php [hunk 3]
```diff
@@ -99,4 +99,37 @@ public function create($data, $invoiceId, $invoiceItemId = null)
...
+    {
+        return $this->client()->put(
+            $data,
+            'invoices/{invoiceId}/items/{invoiceItemId}',
+            ['invoiceId' => $invoiceId, 'invoiceItemId' => $invoiceItemId]
+        );
+    }
+
+    /**
+     * @param array|JsonSerializable|InvoiceItem $data
```
Comment (VladimirBerdnik) [inline src/Services/InvoiceItemService.php:122]:
```suggestion
```

File: src/Services/InvoiceItemService.php [hunk 4]
```diff
@@ -99,4 +99,36 @@ public function create($data, $invoiceId, $invoiceItemId = null)
...
+            'invoices/{invoiceId}/items/{invoiceItemId}',
+            ['invoiceId' => $invoiceId, 'invoiceItemId' => $invoiceItemId]
+        );
+    }
+
+    /**
+     * @param string $invoiceId
+     * @param string $invoiceItemId
+     *
+     * @return InvoiceItem
```
Comment (subenson) [inline src/Services/InvoiceItemService.php:125]:
```suggestion
```

File: src/Services/InvoiceItemService.php (hunk 2 above)
Comment (subenson) [reply src/Services/InvoiceItemService.php]: Good catch. Thanks

File: src/Services/InvoiceItemService.php [hunk 5]
```diff
@@ -99,4 +99,34 @@ public function create($data, $invoiceId, $invoiceItemId = null)
...
+        );
+    }
+
+    /**
+     * @param string $invoiceId
+     * @param string $invoiceItemId
+     */
+    public function delete($invoiceId, $invoiceItemId)
+    {
+        return $this->client()->delete(
```
Comment (ashkarpetin) [inline src/Services/InvoiceItemService.php:127]:
```suggestion
        $this->client()->delete(
```

**Review by ashkarpetin — APPROVED** [review]

**Review by slavcodev — APPROVED** [review]

**Review by XaosSintez — APPROVED** [review]

**Review by vykhrystiuk — APPROVED** [review]

**Pull request merged**